m3u8/writer.go

```diff
@@ -603,6 +684,70 @@ func (p *MediaPlaylist) AppendSegment(seg *MediaSegment) error {
 	return nil
 }
 
+func (p *MediaPlaylist) AppendPartial(uri string, duration float64, independent bool) error {
```

Comment:
Exported methods should have a doc string, especially when not trivial.
This one should look something like:

// AppendPartial creates and and appends a partial segment to p